### PR TITLE
[#9042] Student: disable functionality to add comment in frontend if student has not answer question

### LINKED ISSUE #9042: Student: disable functionality to add comment in frontend if student has not answer question
Due to our database design constraint, a comment from feedback participant should be associated with its corresponding feedback response. This implies: student should not be able to add comment if he/she has not answer the question yet.

The frontend UI currently enable this behaviour but it is just that the backend will ignore the comments from students.
https://github.com/TEAMMATES/teammates/blob/c8ed16c1d43da5c3ecc9b77b65aeb0e8278c29cd/src/test/java/teammates/test/cases/browsertests/InstructorFeedbackSubmitPageUiTest.java#L326-L340

### PR DESCRIPTION
**Pull request by nidhi98gupta**
Fixes #9042

### CODE REVIEW

File: src/main/webapp/dev/js/main/feedbackSubmissionsEdit.js [hunk 1]
```diff
@@ -153,6 +153,47 @@ function focusModeratedQuestion() {
...
+function checkResponsesForFeedbackParticipantComment(indexSuffix) {
+    const radioButton = $(`input[name='responsetext${indexSuffix}']:checked`);
+
+    if (!radioButton.length) {
+        hideFeedbackParticipantCommentOption(indexSuffix);
+    } else if (radioButton.data('text') === 'otherOptionText'
+        && $(`#otherOptionText${indexSuffix}`).val().length === 0) {
+        hideFeedbackParticipantCommentOption(indexSuffix);
+
+        $(`#otherOptionText${indexSuffix}`).keyup(function () {
```
Comment (nidhi98gupta) [inline src/main/webapp/dev/js/main/feedbackSubmissionsEdit.js]: This line is giving me lint error.   `185:51  error  Unexpected function expression  prefer-arrow-callback`.
I am not sure what I am doing wrong.

**Comment by nidhi98gupta** [comment]
@xpdavid Please review js functions that I added. Also java tests which tested comments without responses are ignored should be removed in this PR right?

**Comment by xpdavid** [comment]
> tested comments without responses are ignored should be removed in this PR right?

Need to verify that in the UI, comment box will not be displayed if no response is added.

File: src/main/webapp/dev/js/main/feedbackSubmissionsEdit.js (hunk 1 above)
Comment (nidhi98gupta) [reply src/main/webapp/dev/js/main/feedbackSubmissionsEdit.js]: @LuLechuan Can you help me with this?

**Comment by xpdavid** [comment]
Can you share your idea on how you detect if user enter answer for a response or not?

**Comment by damithc** [comment]
The alternative is to find a way represent 'empty response' in the backend so that even comments on empty responses can be saved, which side-steps this problem altogether.

**Comment by nidhi98gupta** [comment]
Can we we assign response id explicitly (since it is not generated) to comment without responses?

**Comment by xpdavid** [comment]
> Can we we assign response id explicitly (since it is not generated) to comment without responses?

Good idea but it seems like the logic for instructor side would be complicated.

**Comment by damithc** [comment]
> Good idea but it seems like the logic for instructor side would be complicated.

Yes, I guess it's not easy.

**Pull request closed without merging**